model/exclusionList/ExclusionListClassService.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+namespace oat\taoClientDiagnostic\model\exclusionList;
+
+use core_kernel_classes_Class;
+use core_kernel_classes_Property;
+use core_kernel_classes_Resource;
+use oat\generis\model\OntologyRdfs;
+use oat\tao\model\OntologyClassService;
+
+/**
+ * Class exclusionListClassService
+ *
+ * @package oat\taoClientDiagnostic\model\exclusionList
+ */
+abstract class ExclusionListClassService extends OntologyClassService
+{
+    /** @var array */
+    private $names = [];
+
+    /** @var array */
+    private $excluded = [];
+
+    abstract protected function getListClass(): core_kernel_classes_Class;
+
+    abstract public function getNameProperty(): core_kernel_classes_Property;
+
+    abstract public function getNamePropertyUri(): string;
+
+    abstract public function getVersionProperty(): core_kernel_classes_Property;
+
+    abstract public function getVersionPropertyUri(): string;
+
+    public function getListNames(): array
+    {
+        if (!$this->names) {
```

Comment:
should we be able to tell empty exclusion lists from uninitialised lists (NULL maybe?)

Same for `$this->excluded`